src/blockdata/script.rs

```diff
@@ -1105,6 +1162,18 @@ mod test {
         assert_eq!(original, des);
     }
 
+    #[test]
+    #[cfg(feature = "serde")]
+    fn color_id_json_serialize() {
+        use serde_json;
+        let bytes = Vec::from_hex("c3ec2fd806701a3f55808cbec3922c38dafaa3070c48c803e9043ee3642c660b46").unwrap();
+        let color_identifier: ColorIdentifier = deserialize(&bytes).unwrap();
+        let json = serde_json::to_value(&color_identifier).unwrap();
+        assert_eq!(json, serde_json::Value::String("c3ec2fd806701a3f55808cbec3922c38dafaa3070c48c803e9043ee3642c660b46".to_owned()));
+        let des = serde_json::from_value(json).unwrap();
+        assert_eq!(color_identifier, des);
```

Comment:
It is good to have a case when data that does not match the color identifier is read.

Reply:
Add error cases and fix error handling
- invalid token type (0xC0)
- json string is too long/short